### PR TITLE
tests: fix flaky logger tests

### PR DESCRIPTION
**Pull request by JadenSimon**
## Problem
Logger tests aren't consistent. See #2095

## Solution
Wait for streams to flush prior to checking results. 

Note that winston's `finish` event fires only after its internal streams have flushed, not necessarily that the underlying stream has closed. There's currently no way to access the underlying stream unfortunately. Could upstream a fix to winston to correctly emit a 'closed' event after all file system resources have been relinquished, but that's quite a bit of work. So right now polling is still the best solution.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by JadenSimon** [comment]
Need to do #2398 next

File: src/shared/logger/winstonToolkitLogger.ts [hunk 1]
```diff
@@ -107,12 +107,14 @@ export class WinstonToolkitLogger implements Logger, vscode.Disposable {
...
-            this.logger.close()
-            this.logger.clear()
-            this.disposed = true
-        }
+    public dispose(): Promise<void> {
+        return this.disposed
+            ? Promise.resolve()
+            : new Promise<void>(resolve => {
+                  this.disposed = true
+                  this.logger.once('finish', resolve)
```
Comment (justinmk3) [inline src/shared/logger/winstonToolkitLogger.ts:115]: Can't find any docs on this "finish" event. This is a non-obvious line of code, worth adding a 1-line comment explaining the intent + a reference.

```suggestion
                  // Trigger winston "finish" event manually. https://example.com/foo/bar #2397
                  this.logger.once('finish', resolve)
```

**Review by justinmk3 — APPROVED** [review]
LGTM after above comment

File: src/shared/logger/winstonToolkitLogger.ts (hunk 1 above)
Comment (JadenSimon) [reply src/shared/logger/winstonToolkitLogger.ts:116]: The event is from Node's Stream API which is why winston docs doesn't explicitly mention it. Basically winston (even the logger) is just a wrapper around streams, so many of the same ideas apply.

Oh also `once` is registering an event listener, not triggering anything. Just a Node pattern.

**Review by justinmk3 — APPROVED** [review]

**Pull request merged**